File: javascript/lesson 6/postOrder.js
// https://leetcode.com/problems/binary-tree-postorder-traversal/

function postorderTraversal(root) {
    const result = [];
    postOrder(root, result);
    return result;
}

function postOrder(node, result) {
    if (!node) return;

    postOrder(node.left, result);
    postOrder(node.right, result);
    result.push(node.val);
}

File: kotlin/lesson 6/PostOrder.kt
// https://leetcode.com/problems/binary-tree-postorder-traversal/

class Solution {
    fun postorderTraversal(root: TreeNode?): List<Int> {
        val result = mutableListOf<Int>()
        postOrder(root, result)
        return result
    }
    
    fun postOrder(node: TreeNode?, result: MutableList<Int>) {
        if (node == null) return

        postOrder(node.left, result)
        postOrder(node.right, result)
        result.add(node.`val`)
    }
}
